logstash-core/spec/logstash/api/commands/default_metadata_spec.rb

```diff
@@ -12,9 +12,27 @@
 
   let(:report_class) { described_class }
 
+  after :each do
+    LogStash::SETTINGS.register(LogStash::Setting::Boolean.new("xpack.monitoring.enabled", false)) unless LogStash::SETTINGS.registered?("xpack.monitoring.enabled")
+    LogStash::SETTINGS.reset
```

Comment:
`LogStash::SETTINGS` is a singleton instance in the LogStash module, the `reset` operation nullify all the values in the collection, perhaps could be best to have a logic of save-before test restore-after for only the settings used by this test